test/unit/math/fwd/fun/value_of_test.cpp

```diff
@@ -0,0 +1,115 @@
+#include <stan/math/fwd.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+
+TEST(AgradFwd, value_of) {
+  using stan::math::fvar;
+  using stan::math::value_of;
+
+  fvar<double> a = 5.0;
+  EXPECT_FLOAT_EQ(5.0, value_of(a));
+  // make sure all work together
+  EXPECT_FLOAT_EQ(5.0, value_of(5.0));
+  EXPECT_FLOAT_EQ(5.0, value_of(5));
+}
+
+TEST(AgradFwd, value_of_nan) {
+  using stan::math::fvar;
+  using stan::math::value_of;
+  double nan = std::numeric_limits<double>::quiet_NaN();
+
+  fvar<double> a = nan;
+  EXPECT_TRUE(stan::math::is_nan(value_of(a)));
+  EXPECT_TRUE(stan::math::is_nan(value_of(nan)));
+}
+
+TEST(MathMatrixFwdArr, value_of) {
+  using stan::math::fvar;
+  using stan::math::value_of;
+  using std::vector;
+
+  vector<double> a_vals;
+  for (size_t i = 0; i < 10; ++i)
+    a_vals.push_back(i + 1);
+
+  vector<double> b_vals;
+  for (size_t i = 10; i < 15; ++i)
+    b_vals.push_back(i + 1);
+
+  vector<fvar<double> > a_fd;
```

Comment:
[optional] Should we change `> >` to `>>` in this file?

Reply:
Lets tackle this one separately. I do think it improves the code style. But a quick grep shows roughly a hundred files with the same style. This PR is large enough I think.

Reply:
Fair enough!